### PR TITLE
Docs: supported agencies, enrollment pathways

### LINKED ISSUE #1791: Create docs page describing live / active agencies and flows
Idea from @indexing as an additional way to document what features we have live.

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #1791 

## What this PR does
- Adds table to show what transit agencies use what enrollment pathways, with year launched - in the Project overview section
- Adds table to show what enrollment pathways are available, and their general status - in the Project overview section
- Changes `Use cases` to `Enrollment pathways`
- Removes College discount
- Renames `Seniors` to `Older adults`

## Project overview
image

## Sidebar
image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@indexing What do you think?

image

In the future, when new enrollment pathways are planned out - we can add them and add it as a status of  "In contracting", "In product/design", "In development".

**Comment by indexing** [comment]
@machikoyasuda I have a few small suggestions.
Image Dec 12 at 11 AM

**Comment by machikoyasuda** [comment]
@indexing 
image

- Updated `in production validation` to just `in test` for SBMTD/Mobility Pass. I think `In test` could be used to describe the various states between `the code is done and deployed but we are waiting on agencies/permissions/contracts/politics`.
- Is 10/2023 launch for SBMTD/Older adults correct(??) I forgot.
- I added links to the specific release notes for each enrollment pathway launch.

**Comment by indexing** [comment]
_The Cal-ITP Benefits app supports the following enrollment pathways that ~~uses~~ use the corresponding eligibility verification methods._

I note a small verb from typo. Otherwise, looks good! The left alignment of column content _and_ the heading is a cleaner look.

**Comment by machikoyasuda** [comment]
@indexing fixed: https://github.com/cal-itp/benefits/pull/1815/commits/62e1b8f76a43946b68905a5fe357623efb94e6f5

File: docs/README.md
Comment (indexing) [inline docs/README.md:27]: A small formatting suggestion for styling and legibility: represent single months with two digits rather than one, i.e. 05 rather than 5. All date with read cleanly together.

**Review by indexing — APPROVED** [review]

File: docs/README.md
Comment (machikoyasuda) [inline docs/README.md:27]:
```suggestion
| **Monterey-Salinas Transit**                    | Live         | Live        | Live     | 05/2021                |
```

File: docs/README.md
Comment (machikoyasuda) [inline docs/README.md:37]:
```suggestion
| [**Older adults**](/benefits/enrollment-pathways/older-adults) | [Login.gov IAL2](https://developers.login.gov/attributes/)                            | Live   | [08/2022](https://github.com/cal-itp/benefits/releases/tag/2022.08.1)  |
```

File: docs/README.md
Comment (machikoyasuda) [inline docs/README.md:39]:
```suggestion
| [**Veterans**](/benefits/enrollment-pathways/veterans)         | [Veteran Confirmation API](https://developer.va.gov/explore/api/veteran-confirmation) | Live   | [09/2023](https://github.com/cal-itp/benefits/releases/tag/2023.09.1)  |
```

**Comment by machikoyasuda** [comment]
Needs final review from @angela-tran (codeowner)

File: docs/README.md
Comment (machikoyasuda) [reply docs/README.md]: Resolved: https://github.com/cal-itp/benefits/pull/1815/commits/d1281e88476c816d90f805f65eb0d67ec44ff11c https://github.com/cal-itp/benefits/pull/1815/commits/4468bafce04f37baaab2b020ab184959854a428d

File: docs/README.md
Comment (angela-tran) [inline docs/README.md:29]: I'm wondering if we might want to track more than just the initial launch date, as in track the dates that a specific enrollment pathway was launched for a specific transit agency. So the table would be more like:
`| Transit agency | Enrollment pathway | Launch date |`

I think what you have here is a good start though, so I'll leave this as a thought that can be a future PR if we want

**Review by angela-tran — APPROVED** [review]

File: docs/README.md
Comment (machikoyasuda) [reply docs/README.md:29]: Yea we could but that's so granular, and it kinda really only affects MST - since most agencies don't offer a Veterans discount. We're trying to phase out Agency Cards, and perhaps in the future, more agencies can go live with multiple enrollment pathways so we don't have to be doing these incremental rollouts.

**Pull request merged**
